test/plugin/test_out_elasticsearch.rb

```diff
@@ -954,10 +1001,10 @@ def test_connection_failed_retry
       raise Faraday::ConnectionFailed, "Test message"
     end
 
-    driver.emit(sample_record)
-
-    assert_raise(Fluent::ElasticsearchOutput::ConnectionFailure) {
-      driver.run
+    assert_raise(Fluent::Plugin::ElasticsearchOutput::ConnectionFailure) {
```

Comment:
I understand your point regarding hard-coded sleep. The problem is that the retry logic incorporates a backoff, and it seems even when specifying immediate flush that it somehow does not wait automatically for the sleep from the retry.

I don't think removing the assertion is correct, because then we will be missing half the logic of the test. We will know that it tried to reconnect 3 times, but we don't know that after that a ConnectionFailure is thrown. I guess this is not the end of the world, since it is sort of implied.